src/main/java/io/cryostat/net/web/http/api/v1/RecordingUploadPostHandler.java

```diff
@@ -122,36 +120,40 @@ public boolean isAsync() {
 
     @Override
     public void handleAuthenticated(RoutingContext ctx) throws Exception {
-        String recordingName = ctx.pathParam("recordingName");
-        try {
-            URL uploadUrl = new URL(env.getEnv(Variables.GRAFANA_DATASOURCE_ENV));
-            boolean isValidUploadUrl =
-                    new UrlValidator(UrlValidator.ALLOW_LOCAL_URLS).isValid(uploadUrl.toString());
-            if (!isValidUploadUrl) {
-                throw new HttpException(
-                        501,
-                        String.format(
-                                "$%s=%s is an invalid datasource URL",
-                                Variables.GRAFANA_DATASOURCE_ENV, uploadUrl.toString()));
-            }
-            ResponseMessage response = doPost(recordingName, uploadUrl);
-            if (!HttpStatusCodeIdentifier.isSuccessCode(response.statusCode)
-                    || response.statusMessage == null
-                    || response.body == null) {
-                throw new HttpException(
-                        512,
-                        String.format(
-                                "Invalid response from datasource server; datasource URL may be"
-                                    + " incorrect, or server may not be functioning properly: %d"
-                                    + " %s",
-                                response.statusCode, response.statusMessage));
-            }
-            ctx.response().setStatusCode(response.statusCode);
-            ctx.response().setStatusMessage(response.statusMessage);
-            ctx.response().end(response.body);
-        } catch (MalformedURLException e) {
-            throw new HttpException(501, e);
-        }
+        ctx.response()
+                .putHeader(HttpHeaders.LOCATION, "recordings/:sourceTarget/:recordingName/upload");
+        ctx.response().setStatusCode(301).end("ERROR: This endpoint is deprecated.");
```

Comment:
Check out https://github.com/cryostatio/cryostat/pull/1072 . I just added support for deprecating API endpoints to it. Rather than replacing the handler implementation like this with a repeated pattern of `301 / Location`, you just do this:

```java
@DeprecatedApi(deprecated = @Deprecated(forRemoval = true), alternateLocation = "/none")
class HealthGetHandler implements RequestHandler {
```

Obviously the `HealthGetHandler` shouldn't actually be deprecated, I just used that one for testing.

Reply:
When `forRemoval=true`:

```
$ http :8181/health
HTTP/1.1 410 Gone
content-encoding: gzip
content-length: 47
deprecation: true
link: /none; rel=alternate
location: /none
vary: origin

Deprecated. See /none
```

and when `forRemoval=false` (or not set):

```
$ http :8181/health
HTTP/1.1 200 OK
content-encoding: gzip
content-length: 148
content-type: application/json
deprecation: true
link: /none; rel=alternate
vary: origin

{
    "cryostatVersion": "v2.0.0-SNAPSHOT-388-gc1f0124c-dirty",
    "dashboardAvailable": true,
    "dashboardConfigured": true,
    "datasourceAvailable": true,
    "datasourceConfigured": true,
    "reportsAvailable": true,
    "reportsConfigured": true
}
```